server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -83,7 +83,14 @@ public final void handleRequest(RestRequest request, RestChannel channel, NodeCl
         // check if the query has any parameters that are not in the supported set (if declared)
         Set<String> supported = allSupportedParameters();
         if (supported != null) {
-            var allSupported = Sets.union(RestResponse.RESPONSE_PARAMS, ALWAYS_SUPPORTED, supported);
+            var allSupported = Sets.union(
+                RestResponse.RESPONSE_PARAMS,
+                ALWAYS_SUPPORTED,
+                // these internal parameters cannot be set by end-users, but are used by Elasticsearch internally.
+                // they must be accepted by all handlers
+                RestRequest.INTERNAL_MARKER_REQUEST_PARAMETERS,
```

Comment:
we need this because we now set the parameter for actions for which we did _not_ set it before (we didn't set the parameter for internal actions previously, now we do). A couple of these actions (e.g., `RestDeleteSnapshotAction`) override `allSupportedParameters` which in turns means that we run a check against all request parameters and reject "unsupported" ones. The default (for virtually all actions) is to return `null` which means so check, so we simply didn't exercise this code path before.